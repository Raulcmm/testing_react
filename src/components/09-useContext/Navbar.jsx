import React from "react";
import { Link, NavLink } from "react-router-dom";

const Navbar = () => {
	return (
		<nav className="navbar navbar-expand-lg navbar-dark bg-dark">
			<div className="container-fluid">
				<div className="collapse navbar-collapse" id="navbarSupportedContent">
					<ul className="navbar-nav me-auto mb-2 mb-lg-0">
						<li className="nav-item">
							<NavLink
								exact
								to="/"
								activeClassName="active"
								className=" nav-item nav-link "
							>
								Home
							</NavLink>
						</li>
						<li className="nav-item">
							<NavLink
								exact
								to="/about"
								activeClassName="active"
								className=" nav-item nav-link "
							>
								About
							</NavLink>
						</li>
						<li className="nav-item">
							<NavLink
								exact
								to="/login"
								activeClassName="active"
								className=" nav-item nav-link "
							>
								Login
							</NavLink>
						</li>
					</ul>
				</div>
			</div>
		</nav>
	);
};

export default Navbar;
