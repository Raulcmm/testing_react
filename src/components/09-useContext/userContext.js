import React, { createContext, useState } from "react";

export const userContext = createContext(null);

const UserContextProvider = (props) => {
	const [user, setUser] = useState({
		id: 1,
		name: "Raul Camacho",
	});

	return (
		<userContext.Provider value={{ user, setUser }}>
			{props.children}
		</userContext.Provider>
	);
};

export default UserContextProvider;
