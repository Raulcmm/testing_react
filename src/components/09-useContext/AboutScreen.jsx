import React, { useContext } from "react";
import { userContext } from "./userContext";

const AboutScreen = () => {
	const { user, setUser } = useContext(userContext);
	const handleClick = () => {
		setUser({});
	};

	return (
		<div>
			<h1>AboutScreen</h1>
			<hr />
			<pre>{JSON.stringify(user, null, 3)}</pre>
			<h2>{user.name}</h2>

			<button className="btn btn-warning" onClick={handleClick}>
				delete data
			</button>
		</div>
	);
};

export default AboutScreen;
