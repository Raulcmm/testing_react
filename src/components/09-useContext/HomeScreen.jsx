import React, { useContext } from "react";

import { userContext } from "./userContext";
const HomeScreen = () => {
	const { user } = useContext(userContext);
	// console.log(user);
	return (
		<div>
			<h1>HomeScreen</h1>
			<hr />

			<pre>{JSON.stringify(user)}</pre>
		</div>
	);
};

export default HomeScreen;
