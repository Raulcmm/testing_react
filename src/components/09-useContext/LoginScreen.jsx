import React, { useContext } from "react";
import { userContext } from "./userContext";

const LoginScreen = () => {
	const { setUser } = useContext(userContext);
	return (
		<div>
			<h1>LoginScreen</h1>
			<hr />
			<button
				className="btn btn-primary"
				onClick={() => setUser({ id: 123, name: "Osvaldo Camacho" })}
			>
				Agregar datos
			</button>
		</div>
	);
};

export default LoginScreen;
