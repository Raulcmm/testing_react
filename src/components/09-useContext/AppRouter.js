import React from "react";
import { BrowserRouter as Router, Redirect, Route, Switch } from "react-router-dom";
import AboutScreen from "./AboutScreen";
import HomeScreen from "./HomeScreen";
import LoginScreen from "./LoginScreen";
import Navbar from "./Navbar";
const AppRouter = () => {
	return (
		<Router>
			<Navbar />
			<div className="container">
				<Switch>
					<Route exact path="/" component={HomeScreen} />
					<Route exact path="/login" component={LoginScreen} />
					<Route exact path="/about" component={AboutScreen} />
					<Redirect to="/" />
				</Switch>
			</div>
		</Router>
	);
};

export default AppRouter;
