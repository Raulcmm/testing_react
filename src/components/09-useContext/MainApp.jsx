import React from "react";
import AppRouter from "./AppRouter";
import UserContextProvider from "./userContext";

const MainApp = () => {
	return (
		<UserContextProvider>
			<AppRouter />
		</UserContextProvider>
	);
};

export default MainApp;
