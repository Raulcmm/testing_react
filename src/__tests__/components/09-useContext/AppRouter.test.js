import { mount } from "enzyme";
import React from "react";
import AppRouter from "../../../components/09-useContext/AppRouter";
import { userContext } from "../../../components/09-useContext/userContext";

describe("Test to AppRouter", () => {
	const user = { id: 1, name: "Raul" };
	const wrapper = mount(
		<userContext.Provider value={{ user }}>
			<AppRouter />
		</userContext.Provider>
	);

	test("Match con el snap", () => {
		expect(wrapper).toMatchSnapshot();
	});
});
