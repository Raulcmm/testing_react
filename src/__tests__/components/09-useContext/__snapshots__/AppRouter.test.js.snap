// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to AppRouter Match con el snap 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className=" nav-item nav-link  active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className=" nav-item nav-link "
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className=" nav-item nav-link "
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {"id":1,"name":"Raul"}
      </pre>
    </div>
  </div>,
]
`;
